x/subscription/spec/03_messages.md

```diff
@@ -0,0 +1,61 @@
+<!--
+order: 3
+-->
+
+# Messages
+
+## MsgCreatePlan
+
+Merchant create plans, the message signer is the plan owner.
+
+```protobuf
+message MsgCreatePlan {
+  string title;
+  string description;
+  Coin price,  // amount + denomination
+  int32 expiration_time,
+  bool tolerate_failure,
+  int32 interval
+  int32 interval_offset
+}
+```
+
+## MsgStopPlan
+
+Sent by plan owners, also removes all corresponding subscriptions.
+
+```protobuf
+message MsgStopPlan {
+  int32 plan_id,
+}
+```
+
+## MsgCreateSubscription
+
+The message signer subscribe to the plan, fails if it'll expire before next collection time.
+
+```protobuf
+message MsgCreateSubscription {
+  int32 plan_id,
+  int32 expiration_time,
+}
+```
+
+It'll consume some gas for estimated computational cost in future:
+
+```
+period0 = (block_time + interval_offset) / interval
+periodn = (expiration_time + interval_offset) / interval
+ConsumeGas( (periodn - period0 + 1) * GasPerCollection )
+```
+
+## MsgStopSubscription
+
+A subscriber or a plan owner can stop a subscription.
+
+```protobuf
+message MsgStopSubscription {
```

Comment:
If a user has a monthly subscription and they stop a subscription. Does the subscription gets immediately stopped or at the end of month (when they have already paid for the whole month)?

Reply:
It's immediately removed from the state, but if you query the events of success payment collection of current period index, it's still in that list, I think merchant can use that list as the valid subscription list.

Reply:
We don't have a builtin way to query events directly though, unless using chain-indexing.